internal/partinfo/partinfo.go

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) 2023 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package partinfo
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+)
+
+type FSType string
+
+const (
+	FAT32 FSType = "fat32"
+	Ext   FSType = "ext"
+)
+
+type PartInfo interface {
```

Comment:
Normally in Go an interface would only be defined where it's used (i.e., where you're depending on the thing), with the subset of methods that you need. So this package would define `struct Partition` and then callers could use that directly (or define an interface and let `Partition` fulfil that if they needed to).

Interestingly, here `partition` doesn't even seem to implement this interface. Is that intentional?

Additionally, a naming thing: normally in Go you don't use the `Get` prefix on method names, so for example these would be `DevNode()` etc.